pkg/promtail/targets/tailer.go

```diff
@@ -129,12 +129,14 @@ func (t *tailer) stop() error {
 	if err != nil {
 		level.Error(t.logger).Log("msg", "error getting tail position", "path", t.path, "error", err)
 	}
-	err = t.tail.Stop()
+	if err = t.tail.Stop(); err != nil {
+		return err
```

Comment:
Can you explain a little more detail on this change?
I believe if we return here the filesActive count will become inaccurate